In [2]:
import numpy as np
import pandas as pd
import plotly.express as px


In [3]:
log = pd.read_csv("./data_files/logs.csv")

/var/folders/nb/20h7pxd97n9dly3_d0vghc280000gn/T/ipykernel_89234/2326660139.py:1: DtypeWarning: Columns (34,38,39,40,47,50,53,54,57,58,59,60,62,68,70,72,73,74,76,77,78,81,85,88,89,90,91,92,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,118,120,121,122,123,129,131) have mixed types. Specify dtype option on import or set low_memory=False.
  log = pd.read_csv("./data_files/logs.csv")


In [5]:
log

row_id  player_id  school                     wave     session  \
0              1    6427001    3561  3561 (T,TH) Spring 2013   Session 1   
1              2    6427001    3561  3561 (T,TH) Spring 2013   Session 1   
2              3    6427001    3561  3561 (T,TH) Spring 2013   Session 1   
3              4    6427001    3561  3561 (T,TH) Spring 2013   Session 1   
4              5    6427001    3561  3561 (T,TH) Spring 2013   Session 1   
...          ...        ...     ...                      ...         ...   
2106592  2106593   65670032    1531              Single wave  Session 10   
2106593  2106594   65670032    1531              Single wave  Session 10   
2106594  2106595   65670032    1531              Single wave  Session 10   
2106595  2106596   65670032    1531              Single wave  Session 10   
2106596  2106597   65670032    1531              Single wave  Session 10   

               date  event_id                  event_description  \
0        2013-03-07       600                 Player selects age   
1        2013-03-07       601              Player selects gender   
2        2013-03-07       602   Player commits to age and gender   
3        2013-03-07       603              Player selects avatar   
4        2013-03-07       604  Player commits to avatar and name   
...             ...       ...                                ...   
2106592  2013-12-13       706               Player pans epilogue   
2106593  2013-12-13       705        Player closes Epilogue Item   
2106594  2013-12-13         1                Player ends session   
2106595  2013-12-13       706               Player pans epilogue   
2106596  2013-12-13       708            Player hits done button   

          event_category event_time  ...  piece2_id  piece3_id  \
0        Avatar Creation     3M 18S  ...        NaN        NaN   
1        Avatar Creation     3M 19S  ...        NaN        NaN   
2        Avatar Creation     3M 25S  ...        NaN        NaN   
3        Avatar Creation     3M 30S  ...        NaN        NaN   
4        Avatar Creation     4M 12S  ...        NaN        NaN   
...                  ...        ...  ...        ...        ...   
2106592         Epilogue  1H 36M 1S  ...        NaN        NaN   
2106593         Epilogue  1H 36M 5S  ...        NaN        NaN   
2106594          General  1H 36M 7S  ...        NaN        NaN   
2106595         Epilogue  1H 36M 7S  ...        NaN        NaN   
2106596         Epilogue  1H 36M 7S  ...        NaN        NaN   

        opponent_point_loss player_points opponent_points  old_rating  \
0                       NaN           NaN             NaN         NaN   
1                       NaN           NaN             NaN         NaN   
2                       NaN           NaN             NaN         NaN   
3                       NaN           NaN             NaN         NaN   
4                       NaN           NaN             NaN         NaN   
...                     ...           ...             ...         ...   
2106592                 NaN           NaN             NaN         NaN   
2106593                 NaN           NaN             NaN         NaN   
2106594                 NaN           NaN             NaN         NaN   
2106595                 NaN           NaN             NaN         NaN   
2106596                 NaN           NaN             NaN         NaN   

         new_rating  priority_type_that_went_to_zero  epilogue_item_id  \
0               NaN                              NaN               NaN   
1               NaN                              NaN               NaN   
2               NaN                              NaN               NaN   
3               NaN                              NaN               NaN   
4               NaN                              NaN               NaN   
...             ...                              ...               ...   
2106592         NaN                              NaN               NaN   
2106593         Na

In [4]:
player1_log = pd.read_csv("./data_files/player-6427031 copy.csv")
player1_log

/var/folders/nb/20h7pxd97n9dly3_d0vghc280000gn/T/ipykernel_89234/1295355672.py:1: DtypeWarning: Columns (38,39,40,47,88,89,90,91,92,129,131) have mixed types. Specify dtype option on import or set low_memory=False.
  player1_log = pd.read_csv("./data_files/player-6427031 copy.csv")


row_id  player_id  school                         wave    session  \
0     308289    6427031    3561  3561 YES! Program Fall 2013  Session 1   
1     308290    6427031    3561  3561 YES! Program Fall 2013  Session 1   
2     308291    6427031    3561  3561 YES! Program Fall 2013  Session 1   
3     308292    6427031    3561  3561 YES! Program Fall 2013  Session 1   
4     308293    6427031    3561  3561 YES! Program Fall 2013  Session 1   
...      ...        ...     ...                          ...        ...   
7326  315615    6427031    3561  3561 YES! Program Fall 2013  Session 8   
7327  315616    6427031    3561  3561 YES! Program Fall 2013  Session 8   
7328  315617    6427031    3561  3561 YES! Program Fall 2013  Session 8   
7329  315618    6427031    3561  3561 YES! Program Fall 2013  Session 8   
7330  315619    6427031    3561  3561 YES! Program Fall 2013  Session 8   

            date  event_id                                  event_description  \
0     2013-10-23       600                                 Player selects age   
1     2013-10-23       601                              Player selects gender   
2     2013-10-23       602                   Player commits to age and gender   
3     2013-10-23       603                              Player selects avatar   
4     2013-10-23       604                  Player commits to avatar and name   
...          ...       ...                                                ...   
7326  2013-12-11       815  Player cycles through Opportunity Cards using ...   
7327  2013-12-11       815  Player cycles through Opportunity Cards using ...   
7328  2013-12-11       815  Player cycles through Opportunity Cards using ...   
7329  2013-12-11       815  Player cycles through Opportunity Cards using ...   
7330  2013-12-11       816                                  Player picks card   

       event_category  event_time  ...  piece2_id  piece3_id  \
0     Avatar Creation         12S  ...        NaN        NaN   
1     Avatar Creation         14S  ...        NaN        NaN   
2     Avatar Creation         15S  ...        NaN        NaN   
3     Avatar Creation         18S  ...        NaN        NaN   
4     Avatar Creation         36S  ...        NaN        NaN   
...               ...         ...  ...        ...        ...   
7326   Priority Sense  54H 4M 54S  ...        NaN        NaN   
7327   Priority Sense  54H 4M 55S  ...        NaN        NaN   
7328   Priority Sense  54H 4M 56S  ...        NaN        NaN   
7329   Priority Sense  54H 4M 57S  ...        NaN        NaN   
7330   Priority Sense  54H 5M 53S  ...        NaN        NaN   

     opponent_point_loss player_points opponent_points  old_rating  \
0                    NaN           NaN             NaN         NaN   
1                    NaN           NaN             NaN         NaN   
2                    NaN           NaN             NaN         NaN   
3                    NaN           NaN             NaN         NaN   
4                    NaN           NaN             NaN         NaN   
...                  ...           ...             ...         ...   
7326                 NaN           NaN             NaN         NaN   
7327                 NaN           NaN             NaN         NaN   
7328                 NaN           NaN             NaN         NaN   
7329                 NaN           NaN             NaN         NaN   
7330                 NaN           NaN             NaN         NaN   

      new_rating  priority_type_that_went_to_zero  epilogue_item_id  \
0            NaN                              NaN               NaN   
1            NaN                              NaN               NaN   
2            NaN                              NaN               NaN   
3            NaN                              NaN               NaN   
4            NaN                              NaN               NaN   
...          ...                              ...               ...   
7326         NaN         

In [5]:
player_1_notNull = player1_log.iloc[:, 15:]
player_1_notNull[player_1_notNull.notna() == True & player_1_notNull.columns.str.contains("_id")].select_dtypes(exclude = 'float')


skill_id object_id object_locked object_first_time_unlocked  \
0         NaN       NaN           NaN                        NaN   
1         NaN       NaN           NaN                        NaN   
2         NaN       NaN           NaN                        NaN   
3         NaN       NaN           NaN                        NaN   
4         NaN       NaN           NaN                        NaN   
...       ...       ...           ...                        ...   
7326      NaN       NaN           NaN                        NaN   
7327      NaN       NaN           NaN                        NaN   
7328      NaN       NaN           NaN                        NaN   
7329      NaN       NaN           NaN                        NaN   
7330      NaN       NaN           NaN                        NaN   

     scan_start_point scan_end_point object_scan_result animatic_id  \
0                 NaN            NaN                NaN         NaN   
1                 NaN            NaN                NaN         NaN   
2                 NaN            NaN                NaN         NaN   
3                 NaN            NaN                NaN         NaN   
4                 NaN            NaN                NaN         NaN   
...               ...            ...                ...         ...   
7326              NaN            NaN                NaN         NaN   
7327              NaN            NaN                NaN         NaN   
7328              NaN            NaN                NaN         NaN   
7329              NaN            NaN                NaN         NaN   
7330              NaN            NaN                NaN         NaN   

     avatar_gender avatar_name  ... clip3_validity opponent_attack_id  \
0              NaN         NaN  ...            NaN                NaN   
1              NaN         NaN  ...            NaN                NaN   
2              NaN         NaN  ...            NaN                NaN   
3              NaN         NaN  ...            NaN                NaN   
4              NaN         NaN  ...            NaN                NaN   
...            ...         ...  ...            ...                ...   
7326           NaN         NaN  ...            NaN                NaN   
7327           NaN         NaN  ...            NaN                NaN   
7328           NaN         NaN  ...            NaN                NaN   
7329           NaN         NaN  ...            NaN                NaN   
7330           NaN         NaN  ...            NaN                NaN   

     chosen_opponent_attack_type_id strategy_id piece_id_2 piece1_id  \
0                               NaN         NaN        NaN       NaN   
1                               NaN         NaN        NaN       NaN   
2                               NaN         NaN        NaN       NaN   
3                               NaN         NaN        NaN       NaN   
4                               NaN         NaN        NaN       NaN   
...                             ...         ...        ...       ...   
7326                            NaN         NaN        NaN       NaN   
7327                            NaN         NaN        NaN       NaN   
7328                            NaN         NaN        NaN       NaN   
7329                            NaN         NaN        NaN       NaN   
7330                            NaN         NaN        NaN       NaN   

     piece2_id piece3_id priority_type_that_went_to_zero  \
0          NaN       NaN                             NaN   
1          NaN       NaN                             NaN   
2          NaN       NaN                             NaN   
3          NaN       NaN                             NaN   
4          NaN       NaN                             NaN   
...        ...       ...                             ...   
7326       NaN       NaN                             NaN   
7327       NaN       NaN                             NaN   
7328       NaN       NaN                             

In [6]:
player_1_notNull.dtypes

skill_level_know                   float64
skill_level_priority               float64
skill_level_people                 float64
skill_level_refusal                float64
skill_level_me                     float64
                                    ...   
old_rating                         float64
new_rating                         float64
priority_type_that_went_to_zero     object
epilogue_item_id                   float64
item_locked_unlocked_state          object
Length: 117, dtype: object

In [7]:
pd.options.display.max_rows = 4000
p = player1_log.iloc[:, 15:].notna().sum()
p.sort_values(ascending = False)

event_type                         7331
name_of_file                       7331
minigame_level                     3857
starter_statement_id                915
npc_id                              885
scan_start_point                    712
scan_end_point                      712
priority_type                       611
opportunity_card_id                 519
object_id                           497
card_id                             487
npc_fact_data_type                  480
pan_scene_start_left                426
pan_scene_end_left                  426
object_scan_result                  336
invitation_id                       297
fact_card_id                        289
sense_id                            233
opponent_its_on_statement_id        225
inviting_npc_id                     198
inviting_npc_rank_id                198
aa_level_id                         184
object_first_time_unlocked          161
object_locked                       161
chunk_id                            157


In [8]:
player1_log["event_id"].unique()

array([ 600,  601,  602,  603,  604,  605,  210,  211,  213,  212,  207,
        205,  221,  220,  300,  305,  302,  303,  304,  306,  203,  100,
        102,  200,  219,  216,  217,  218,  204, 1003, 1004,  900,  901,
        904, 1002,  902,  903,  905,  906,  907,  908,  909,  910,  911,
       1005, 1001,  800,  801,  814,  815,  816,  803,  804,  805,  806,
        807,  808,  809,  810,  818,  811,  400,  401,  402,  405,  406,
        411,  403,  404,  412,  413,  414,  415,  416,  417,  418,  419,
        407,  408,  420, 1000,  409,  410,  103,  104,  105,  703,  706,
        707,    1,  708,  500,  501,  502,  503,  504,  505,  506,  507,
        508,  509,  510,  511,  512,  513,  514,  515,  516,  517,  301,
        704,  705,  101,  817,  912,  812,  813])

## Analysing correct choices
| event_category      | event_id           | event_description  | validity |
| ------------- |:-------------:| :-----:| -----:|
| Knowledge Minigame      | 402 | Player selects "True" | true or false |
| Knowledge Minigame      | 403      |  Player selects "False" |  |

In [9]:
# dataframe for validity check on each action

validity_check = log[['player_id', 'date', 'event_id', 'event_category', 'event_description', 'validity']]
validity_check = validity_check[validity_check.validity.notnull()]
# remove ']' string from end of each boolean
validity_check['validity'] = validity_check['validity'].str.strip(']')

validity_check.to_csv('validity_check.csv')
validity_check

player_id        date  event_id      event_category  \
1288       6427001  2013-03-07       403  Knowledge Minigame   
1299       6427001  2013-03-07       408  Knowledge Minigame   
1303       6427001  2013-03-07       402  Knowledge Minigame   
1308       6427001  2013-03-07       404  Knowledge Minigame   
1330       6427001  2013-03-07       404  Knowledge Minigame   
...            ...         ...       ...                 ...   
2106125   65670032  2013-12-13       408  Knowledge Minigame   
2106148   65670032  2013-12-13       404  Knowledge Minigame   
2106170   65670032  2013-12-13       402  Knowledge Minigame   
2106175   65670032  2013-12-13       403  Knowledge Minigame   
2106195   65670032  2013-12-13       408  Knowledge Minigame   

                                         event_description validity  
1288                                Player selects "False"     true  
1299     Player selects "Use Fact" to use chosen fact card    false  
1303                                 Player selects "True"     true  
1308                              Player selects "Opinion"     true  
1330                              Player selects "Opinion"    false  
...                                                    ...      ...  
2106125  Player selects "Use Fact" to use chosen fact card     true  
2106148                           Player selects "Opinion"     true  
2106170                              Player selects "True"     true  
2106175                             Player selects "False"     true  
2106195  Player selects "Use Fact" to use chosen fact card     true  

[48128 rows x 6 columns]

In [10]:
survey = pd.read_csv("./data_files/S5_scores_cleaned.csv")

In [11]:
survey_pivot = survey.pivot_table(values='S5_mean', columns='weeks', index = "player_id")
survey_pivot.columns.name = None
survey_pivot


0    3      6    12   24
player_id                           
6427001.0  3.9  3.9  3.875  NaN  NaN
6427027.0  3.8  4.0  4.000  3.8  3.6
6427030.0  2.9  3.0  3.200  3.3  3.5
6427032.0  4.0  3.9  4.000  3.9  3.8
6427034.0  4.0  3.8  4.000  3.9  4.0
6427036.0  3.8  3.8  3.700  4.0  NaN
6427037.0  4.0  NaN  4.000  4.0  NaN
6427038.0  NaN  NaN    NaN  3.1  NaN
6427039.0  4.0  4.0  4.000  4.0  4.0
6427040.0  3.9  3.8  4.000  3.9  NaN
6427041.0  3.9  3.8  3.000  4.0  3.2
6429001.0  3.8  NaN    NaN  NaN  NaN
6430002.0  4.0  4.0  4.000  4.0  4.0
6430003.0  4.0  4.0  3.700  NaN  NaN
6430004.0  3.9  3.7  3.900  3.8  NaN
6430008.0  4.0  3.7  3.700  4.0  3.9
6486001.0  4.0  NaN    NaN  NaN  3.7
6486002.0  4.0  4.0  3.800  4.0  3.9
6486003.0  4.0  4.0    NaN  4.0  4.0
6486004.0  3.9  3.9  4.000  3.9  NaN
6486005.0  3.5  3.5  3.600  3.7  4.0
6486009.0  4.0  3.1    NaN  3.8  NaN
6486011.0  4.0  4.0  4.000  4.0  NaN
6486013.0  4.0  NaN    NaN  NaN  NaN
6486018.0  1.3  3.7  3.100  NaN  NaN
6486019.0  3.0  4.0  3.900  4.0  NaN
6486020.0  4.0  NaN    NaN  NaN  NaN
6486023.0  4.0  3.9  4.000  4.0  NaN
6486025.0  3.9  NaN  4.000  4.0  NaN
6486026.0  4.0  4.0  4.000  4.0  NaN
6486029.0  4.0  4.0  4.000  3.5  NaN
6486032.0  4.0  4.0  4.000  3.9  NaN
6486033.0  4.0  4.0  4.000  4.0  NaN
6486034.0  3.9  NaN    NaN  NaN  NaN
6486038.0  4.0  4.0  4.000  4.0  NaN
6486039.0  4.0  4.0  4.000  4.0  NaN
6486040.0  4.0  NaN    NaN  3.9  NaN
6506008.0  4.0  NaN  3.900  4.0  3.3
6506009.0  3.9  NaN    NaN  3.9  4.0
6546001.0  3.7  3.9  3.875  3.9  3.8
6546005.0  4.0  3.7  3.700  3.7  3.7
6546006.0  4.0  4.0  4.000  4.0  4.0
6546008.0  3.9  4.0  4.000  NaN  NaN
6546011.0  3.5  3.7  3.800  NaN  NaN
6546014.0  3.9  3.9    NaN  NaN  NaN
6566001.0  4.0  4.0  4.000  NaN  NaN
6566003.0  4.0  4.0    NaN  NaN  NaN
6567002.0  3.6  4.0  4.000  4.0  4.0
6567003.0  4.0  4.0  4.000  4.0  4.0
6567005.0  3.6  NaN  3.100  NaN  NaN
6606001.0  4.0  3.6    NaN  NaN  NaN
6606002.0  3.1  NaN    NaN  NaN  NaN
6606003.0  3.5  3.1  3.800  4.0  4.0
6606004.0  3.9  4.0  4.000  4.0  4.0
6606005.0  3.0  4.0  4.000  4.0  NaN
6606006.0  4.0  4.0    NaN  NaN  NaN
6606011.0  4.0  3.8  3.800  NaN  NaN
6607006.0  4.0  NaN  4.000  2.7  4.0
6607007.0  4.0  4.0  4.000  4.0  4.0
6607008.0  3.9  NaN  3.900  3.9  3.9
6607009.0  4.0  4.0  3.900  4.0  4.0
6607010.0  4.0  4.0  4.000  4.0  4.0
6608002.0  4.0  4.0  3.900  4.0  4.0

In [12]:
survey_scores = pd.read_csv("./data_files/S5_scores_cleaned.csv").fillna(0)
survey_scores['weeks'] = survey_scores['weeks']
fig = px.box(survey_scores, x ='weeks', y="S5_mean", color='weeks', title= 'Boxplot of means scores over the weeks')
fig.update_xaxes(type='category') # hide dates with no values
fig.show()

In [13]:
fig = px.line(survey_scores, x="weeks", y='S5_mean', markers=True, color=str('player_id'), title='Trends of mean scores over the weeks')
fig.show()

### Plot Survey Score to Event time for each participant

In [14]:
# event time is given to us in seconds

#sample last row for each id, from each df

# get ingame timelogs for each participant
timelogs = log.groupby("player_id").event_time_dbl.apply(list).reset_index()

# get final survey response
final_survey = survey.groupby("player_id").last()
final_survey = final_survey[['S5_mean']]

In [15]:
ingametimeSurvey = pd.merge(timelogs, final_survey, on='player_id')
ingametimeSurvey["time_played"] = 0

In [16]:
#Function to calculate the actual time played
def timeCal(df):
  play_start = int(df[0])
  play_end = 0
  total = 0
  #print("Real time: " + str(int(timelogs.iloc[len(timelogs) - 1]) / 3600))

  for i in range(1, len(df)):
    row = int(df[i])
    play_end = row

    if i == (len(df) - 1): 
      total = total + (play_end - play_start)
      #print("Ending calculate: " + str(play_end - play_start))
      break 

    row1 = int(df[i + 1])

    if ((row1 - play_end) / 3600) > 12:
      total = total + (play_end - play_start)
      #print("Calculate: " + str(play_end - play_start))
      play_start = row1
  
  return total

In [17]:
# And drop rows with > 25 hours
for i in range(len(ingametimeSurvey)): 
  actualTime = timeCal(ingametimeSurvey.iloc[i][1])
  #print(actualTime/3600)
  ingametimeSurvey.at[i, "time_played"] = actualTime/3600 

ingametimeSurvey.drop("event_time_dbl", inplace=True, axis=1)
ingametimeSurvey 

player_id  S5_mean  time_played
0     6427001    3.875    10.422222
1     6427027    3.600    10.133611
2     6427030    3.500    10.087500
3     6427032    3.800     1.083889
4     6427034    4.000     7.768889
5     6427036    4.000     9.311111
6     6427037    4.000     8.395000
7     6427038    3.100     8.992222
8     6427039    4.000     3.568333
9     6427040    3.900    11.069722
10    6427041    3.200     9.126111
11    6429001    3.800     3.846389
12    6430002    4.000     8.496944
13    6430003    3.700     6.918056
14    6430004    3.800     8.499167
15    6430008    3.900     1.179167
16    6486001    3.700     7.516944
17    6486002    3.900     7.407778
18    6486003    4.000    10.362500
19    6486004    3.900     9.541389
20    6486005    4.000     7.331667
21    6486009    3.800     8.452778
22    6486011    4.000     2.803333
23    6486013    4.000     8.251667
24    6486018    3.100    10.777778
25    6486019    4.000     9.834722
26    6486020    4.000     8.875833
27    6486023    4.000     6.894167
28    6486025    4.000    10.562778
29    6486026    4.000    10.018056
30    6486029    3.500     7.351944
31    6486032    3.900     8.393056
32    6486033    4.000     6.246389
33    6486034    3.900     8.706944
34    6486038    4.000     8.943333
35    6486039    4.000    10.157500
36    6486040    3.900     8.194722
37    6506008    3.300     9.742778
38    6506009    4.000     5.821111
39    6546001    3.800    10.218889
40    6546005    3.700     9.904722
41    6546006    4.000    11.316111
42    6546008    4.000    10.540556
43    6546011    3.800     6.133611
44    6566001    4.000    10.496111
45    6566003    4.000     8.237500
46    6567002    4.000     6.102778
47    6567003    4.000     6.392778
48    6606001    3.600     2.481111
49    6606002    3.100     3.464722
50    6606003    4.000    11.863611
51    6606004    4.000    11.256111
52    6606005    4.000     8.919444
53    6606006    4.000     7.580278
54    6606011    3.800     2.524722
55    6607006    4.000    10.608889
56    6607007    4.000     2.618889
57    6607008    3.900     7.705000
58    6607009    4.000     0.731944
59    6607010    4.000     8.306111
60    6608002    4.000     8.882778

In [22]:
#plot
playTime_fig = px.scatter(x=ingametimeSurvey.time_played, y=ingametimeSurvey.S5_mean, trendline="ols", trendline_scope="overall")
playTime_fig

In [31]:
results = px.get_trendline_results(playTime_fig)
print(results)
a = px.get_trendline_results(playTime_fig).px_fit_results.iloc[0].rsquared 
print("The R squared of the model is: " + str(round(a, 6)))

                                      px_fit_results
0  <statsmodels.regression.linear_model.Regressio...
The R squared of the model is: 0.000102


## Looking at priorities 

In [20]:
unique = log['priority_type_that_went_to_zero'].dropna().unique()
log['priority_type_that_went_to_zero'] = log['priority_type_that_went_to_zero'].replace(unique, ['Health','Money','School','Friends','Happiness','Family'])
count_priority = log.groupby('priority_type_that_went_to_zero').count().reset_index()#.value_counts()
count_priority

priority_type_that_went_to_zero  row_id  player_id  school  wave  session  \
0                          Family       5          5       5     5        5   
1                         Friends       8          8       8     8        8   
2                       Happiness      48         48      48    48       48   
3                          Health     108        108     108   108      108   
4                           Money      79         79      79    79       79   
5                          School     127        127     127   127      127   

   date  event_id  event_description  event_category  ...  piece1_id  \
0     5         5                  5               5  ...          0   
1     8         8                  8               8  ...          0   
2    48        48                 48              48  ...          0   
3   108       108                108             108  ...          0   
4    79        79                 79              79  ...          0   
5   127       127                127             127  ...          0   

   piece2_id  piece3_id  opponent_point_loss  player_points  opponent_points  \
0          0          0                    0              0                0   
1          0          0                    0              0                0   
2          0          0                    0              0                0   
3          0          0                    0              0                0   
4          0          0                    0              0                0   
5          0          0                    0              0                0   

   old_rating  new_rating  epilogue_item_id  item_locked_unlocked_state  
0           0           0                 0                           0  
1           0           0                 0                           0  
2           0           0                 0                           0  
3           0           0                 0                           0  
4           0           0                 0                           0  
5           0           0                 0                           0  

[6 rows x 132 columns]

In [21]:
fig = px.bar(count_priority, x='priority_type_that_went_to_zero', y='player_id', color="priority_type_that_went_to_zero",
labels={
        "player_id": "Number of times reaching zero",
        "priority_type_that_went_to_zero": "Priority Types",
        "priority_type_that_went_to_zero": "Priority Types"
        } 
)
fig.show()